### ISSUE TITLE
TPAC 2022

### ISSUE THREAD

**Issue by travisleithead** [w3c/editing#400]
As we announced on 11 April, the next TPAC 2022 [1] will be a hybrid meeting with the main in-person hub in Vancouver, Canada:

TPAC 2022
Hybrid Meeting
12-16 September 2022

Main in-person hub:

Sheraton Vancouver Wall Centre
1000 Burrard Street
Vancouver
Canada
British Columbia V6Z 2R9

We will adapt our format to a hybrid mode and plan to build a dynamic meeting to bring more cross participation and new opportunities for work between groups. Together with group and breakout meetings, we hope to organize other sessions such as workshops, hackathons, demos, a developer meetup, an AC meeting...

In order to build the general schedule for the week, we need to know when your groups will meet. To achieve a dynamic meeting and plan other types of sessions, we recommend groups plan shorter sessions for working meetings: from 2-4 hours per day.
We'll also adapt the agenda if some groups plan longer sessions.

Please discuss within your groups and fill in the form below on 31 May at the latest:
https://www.w3.org/2002/09/wbs/1/LogisticsTPAC2022/

We thank you in advance for your input to plan a successful hybrid event!

Best regards,

Alexandra Lacourba
W3C Global Events Coordinator

[1] https://lists.w3.org/Archives/Member/chairs/2022AprJun/0009.html

**Comment by johanneswilm** [comment]
I think it's important to hear whether some representatives who don't usually attend the editing calls will possibly attend our meeting at TPAC in person. Specifically, Firefox and Google. Who, if anyone, is working on editing at Google?

@masayuki-nakano Do you know whether a Mozilla editing engineer will be present in Canada for TPAC?

**Comment by johanneswilm** [comment]
Hey
@BoCupp-Microsoft @whsieh @snianu @alexkeng @saschanaz @yosinch @rakuco @megangardner @Comandeer @garykac @Reinmar @kojiishi @masayuki-nakano @annevk @frivoal,

the W3C needs an answer from the Web Editing Working Group on whether and what kind of meeting we plan to have at TPAC before the 31st of May. My question to you: 

- Do you think you or the organizations you work with/for would participate in an in-person meeting at TPAC this year (2022)? 
- Should we have a 2-hour or 4-hour meeting?
- Alternatively: If we don't have a meeting at TPAC, what alternative would you prefer?

**Comment by Comandeer** [comment]
I won't be able to participate in an in-person meeting, I can attend only remotely. As for the duration of the meeting: 4-hour one sounds a little too long, 2-hour one is perfect.

**Comment by masayuki-nakano** [comment]
@johanneswilm Hi, sorry for the delay to reply. I don't know, and I'll ask, but unfortunately, some countries are in horidays, so I'll get feedback next week, I guess.

**Comment by smaug----** [comment]
I'm interested in participating, but whether in-person or remotely, that I don't know yet. It depends on what else might happen at TPAC. Trying to figure that out asap.

Would of course help if there was at least some preliminary agenda, or proposal for an agenda.

**Comment by johanneswilm** [comment]
The topics that have come up recently include:

- Clipboard API
- EditContext
- Input Events
- Virtual Keyboard API

So there is some chance topics will center around these themes. However, our TPAC meetings tend to draw some participants working with browser engines who do not usually show up at our calls. Particularly Mozilla and Google have not been represented continuously, so this would be a chance to take hear input from those as well.

From the participation of Firefox in the interop discussions concerning editing it sounded like there may also be some proposals on working on standardizing `execCommand` (to some degree), but you, @smaug---- , will probably know better than me if that is a topic Mozilla will want to prepare a proposal for to present at TPAC.

**Comment by johanneswilm** [comment]
We have until Tuesday to give a final answer. Based on the response here so far, I'd say we should not plan an extensive meeting at TPAC 2022 and instead focus on the monthly calls this year and try to get Google/Mozilla involved a little more by looking at ways to make those calls working better for them. Anyone with a different view?

**Comment by whsieh** [comment]
> We have until Tuesday to give a final answer. Based on the response here so far, I'd say we should not plan an extensive meeting at TPAC 2022 and instead focus on the monthly calls this year and try to get Google/Mozilla involved a little more by looking at ways to make those calls working better for them. Anyone with a different view?

This sounds good to me! A 2-hour meeting during TPAC sounds reasonable, as well.

(Separately, I also wonder if it's possible to split the two-hour block into two 1-hour chunks — one that's more friendly to PST + CST, and another for either CST + JST or PST + JST?)

**Comment by johanneswilm** [comment]
> (Separately, I also wonder if it's possible to split the two-hour block into two 1-hour chunks — one that's more friendly to PST + CST, and another for either CST + JST or PST + JST?)

I'll add that in the answer form.

**Comment by snianu** [comment]
Tagging some Google folks to see if they are interested to meet at TPAC. @a-sully @inexorabletash for clipboard. I can only do remote because of travel concerns.

**Comment by inexorabletash** [comment]
I'm planning to attend in person 😷 and would attend sessions on editing topics although primarily as an observer, conflicts permitting.

**Comment by garykac** [comment]
I'll be there in person as well for clipboard-related things.

**Comment by a-sully** [comment]
I'll be there in-person, as well. Happy to attend clipboard-related sessions (conflicts permitting)

**Comment by tilgovi** [comment]
I'm not planning to attend in person, but I would love to attend remotely.

**Comment by zcorpan** [comment]
@howard-e and myself would like to attend remotely.

We are working on specifying and testing `contenteditable=plaintext-only` which is supported in WebKit and Chromium, which involves changing the archived `execCommand` spec (which is normatively referenced by HTML). We'd like to attend one or both of the web editing meetings at TPAC and add to the agenda reviewing our proposed spec change. The relevant spec issue is https://github.com/whatwg/html/issues/7864

I can only attend the first session on Thursday.

**Comment by johanneswilm** [comment]
Hey @zcorpan, that sounds interesting. Have you taken a look at this [1]? That is the latest state of affairs when we discussed other cotnenteditable modes a few years ago.

[1] https://github.com/w3c/contentEditable/blob/old_snapshot/index.html

**Comment by zcorpan** [comment]
@johanneswilm interesting, thanks. Are any of those other than `plaintext-only` implemented in some browser?

**Comment by johanneswilm** [comment]
@zcorpan No, we wrote this at the time with the objective to create a minimal interface for JS devs to create text editors. We were never going to have all the extra modes, but we couldn't figure out which of them would be more useful, or whether it would be enough to just use the pre-existing plaintext-only. We went in circles several times. Then at some point we thought that we should first finish with the Input Events spec and have that implemented everywhere. And then when Chrome decided not to fully implement Input Events, we were waiting for the replacement - which is only now coming in the form of EditContext.

So I think there are two parts that are of interest when it comes to plaintext-only - first there is documenting how it currently works and potentially implement it in browsers that don't yet have it.

The other part is to continue the work from back then and find out whether or not it can be used when beuilding richtext editors in combination with EditContext and/or Input Events, wehther that would still require another mode, or whether other modes are not really needed at all as one can simply preventDefault all the beforeinput events.
